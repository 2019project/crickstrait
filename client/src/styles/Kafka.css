#kafkamainContainer {
  width: 100%;
  display: grid;
  color: #fff;
  padding: 10px 0px;
  background-color: #33485d;
  border: none;
  grid-template-columns: repeat(6, 1fr);
  font-family: Karla;
  text-align: center;
  overflow-x: scroll;
}

#change-ve {
  color: #ff4d4d;
}

#changepositive {
  color: #27ae60;
}

#kafka {
  padding: 0px 10px;
}

#kafka_index_price {
  display: flex;
  justify-content: center;
}

@media screen and (max-width: 430px) {
  #kafkamainContainer {
    width: 100%;
    display: grid;
    color: #fff;
    padding: 10px 0px;
    background-color: #33485d;
    border: none;
    grid-template-columns: repeat(6, 1fr);
    font-family: Karla;
    text-align: center;
    font-size: 14px;
  }
}
