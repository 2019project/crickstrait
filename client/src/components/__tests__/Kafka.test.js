/*kafka component test case */
import React from "react";
import { shallow, mount } from "enzyme";
import Kafka from "../Kafka";

const data = [{ tickerName: "DJI" }];

const wrapper = shallow(<Kafka />);

describe("test for the text, input, css properties and icons on the navbar and snapshot testing", () => {
  it("renders the component for snapshot testing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("checks for the mapping condition", () => {
    wrapper.setState({ data: data });
    // Testing the Div with a particular ID and expecting its Text to be the Declared One[ piysuh]
    expect(wrapper.find("#kafka_indices").text()).toBe("DJI");
  });

  it("checks for onError setState", () => {
    const onError = jest.spyOn(wrapper.instance(), "onError");
    const e = {
      target: {
        error: "Not Found!"
      }
    };
    wrapper.instance().onError(e);
  });

  it("checks for the onConnected setState", () => {
    const onConnected = jest.spyOn(wrapper.instance(), "onConnected");
    const d = {
      target: {
        data: []
      }
    };
    wrapper.instance().onConnected(d);
  });
  //json parse error
  // it("checks for the onMessageRecieved setState", () => {
  //   const onMessageReceived = jest.spyOn(
  //     wrapper.instance(),
  //     "onMessageReceived"
  //   );
  //   let msg = { body: "value" };
  //   // const c = {
  //   //   target: {
  //   //     data: body
  //   //   }
  //   // };
  //   wrapper.instance().onMessageReceived(msg);
  // });
});
