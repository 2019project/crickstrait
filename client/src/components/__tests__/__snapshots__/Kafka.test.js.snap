// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for the text, input, css properties and icons on the navbar and snapshot testing renders the component for snapshot testing 1`] = `
<div
  id="kafkamainContainer"
/>
`;
